common/src/main/java/me/lucko/luckperms/common/treeview/TreeNode.java

```diff
@@ -73,13 +75,13 @@ public TreeNode() {
     }
 
     // lazy init
-    private synchronized Map<String, TreeNode> getChildMap() {
-        if (this.children == null) {
-            this.children = new ConcurrentHashMap<>();
-        }
-        return this.children;
+    private Map<String, TreeNode> getChildMap() {
+        ConcurrentMap<String, TreeNode> a;
+        do {} while ((a = children.get()) == null &&
+                !children.compareAndSet(null,
+                        a = new ConcurrentHashMap<>()));
```

Comment:
Is there any unaesthetic reason to use do while like this?

Reply:
no, absolutely not, you can use a regular while/for loop

Also, starting with jdk 9+, you can do it without a loop at all using compareAndExchange, but we are using java 8